src/domain/opportunity/opportunity.service.ts

```diff
@@ -251,20 +251,51 @@ export class OpportunityService {
     opportunityID: number,
     opportunityData: OpportunityInput
   ): Promise<IOpportunity> {
-    const Opportunity = await this.getOpportunityByID(opportunityID);
+    const opportunity = await this.getOpportunityByID(opportunityID);
 
     // Copy over the received data
     if (opportunityData.name) {
-      Opportunity.name = opportunityData.name;
+      opportunity.name = opportunityData.name;
     }
 
     if (opportunityData.state) {
-      Opportunity.state = opportunityData.state;
+      opportunity.state = opportunityData.state;
+    }
+
+    if (opportunityData.context && opportunity.context) {
+      if (opportunityData.context.background)
+        opportunity.context.background = opportunityData.context.background;
+
+      if (opportunityData.context.impact)
+        opportunity.context.impact = opportunityData.context.impact;
+
+      if (opportunityData.context.references) {
+        for (const referenceInput of opportunityData.context.references) {
+          if (
+            !opportunity.context.references?.some(
```

Comment:
Missing updates to references? This should be just coded once and in the Context object.

Reply:
Fixed.